src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -267,6 +267,11 @@ public void Save(Stream stream)
                 throw new InvalidOperationException(NuGetResources.SemVerSpecialVersionTooLong);
             }
 
+            if (Version != null && Version.IsSemVer2)
+            {
+                throw new InvalidOperationException(String.Format(CultureInfo.CurrentCulture, NuGetResources.SemVer2VersionsNotSupported, Version.ToString()));
```

Comment:
nit: the `Version.ToString()` can be `Version` right? The `.ToString()` should be implicit in `String.Format`.